### PR TITLE
feat: add query parameter for enabling index to v3 position API

### LINKED ISSUE #1189: [FEATURE] Add a query flag to v3 Position API so as to retrieving from indexed data
## Is your feature request related to a problem? Please describe.
Currently, in v3 Position API it takes long time to get balances on an account because it gets data from on-chain token one by one.
From a latency perspective, it may be preferable to be able to use the data collected by the indexer batch as account balance information instead.

> 現状v3 Position APIでは、オンチェーンから1件ずつデータを取得するため、アカウントの残高を取得するのに時間がかかる。
レイテンシを考えると、indexerバッチが収集したデータをアカウント残高の情報として使用できるほうが好ましい場合がある。

## Describe the solution you'd like
We would like to add a query flag to v3 Position API whether to enable indexed data or not.

> v3 Position APIに、indexデータ有効化するためのクエリパラメータを追加する。

### PR DESCRIPTION
**Pull request by purplesmoke05**
Close #1189

### CODE REVIEW

File: app/api/v3/position.py [hunk 1]
```diff
@@ -68,12 +84,80 @@ def on_get_list(self, req, res, account_address=None):
...
+            self.on_success(res, data)
+            return
+
+        # If enable_index flag is not set or set false, get position data from contract.
+        data = self.on_get_list_from_contract(
+            request_json=request_json,
+            session=session,
+            account_address=account_address
+        )
+        self.on_success(res, data)
```
Comment (purplesmoke05) [inline app/api/v3/position.py:107]: Added enable_index flag.
If it is true, API will return data from DB.

File: app/api/v3/position.py [hunk 2]
```diff
@@ -448,6 +587,96 @@ def __init__(self):
...
+                    sub_tx_used.c.used != 0,
+                    session.query(IDXTransfer).filter(
+                        and_(
+                            IDXTransfer.token_address == Listing.token_address,
+                            IDXTransfer.to_address == account_address
+                        )
+                    ).exists()
+                )
+            ).order_by(Listing.id)
+        )
```
Comment (purplesmoke05) [inline app/api/v3/position.py:639]: Unlike other token type, coupon API handles IDXTransfer and returns "used" amount in current.
So sub query is required here.

File: app/model/db/idx_token.py [hunk 3]
```diff
@@ -338,13 +338,13 @@ def json(self):
...
     Type[IDXShareToken],
     Type[IDXBondToken],
     Type[IDXMembershipToken],
     Type[IDXCouponToken]
 ]
-TokenModelInstanceTypes = Union[
+IDXTokenInstance = Union[
     IDXBondToken,
     IDXShareToken,
     IDXMembershipToken,
```
Comment (purplesmoke05) [inline app/model/db/idx_token.py:350]: Renamed.

File: app/api/v3/position.py [hunk 4]
```diff
@@ -440,6 +532,53 @@ def __init__(self):
...
+        query = session.query(Listing.token_address, IDXPosition, self.idx_token_model).\
+            join(self.idx_token_model, Listing.token_address == self.idx_token_model.token_address).\
+            join(IDXPosition, Listing.token_address == IDXPosition.token_address).\
+            filter(IDXPosition.account_address == account_address).filter(
+                or_(
+                    IDXPosition.balance != 0,
+                    IDXPosition.exchange_balance != 0,
+                    IDXPosition.exchange_commitment != 0
+                )
+            ).order_by(Listing.id)
```
Comment (purplesmoke05) [inline app/api/v3/position.py:550]: In Membership Position endpoint, "pending_transfer" is not required.

**Review by YoshihitoAso — APPROVED** [review]

**Pull request merged**